mlir/lib/Dialect/Rock/Transforms/GridwiseGemmToBlockwise.cpp

```diff
@@ -1016,19 +1016,41 @@ struct GridwiseGemmV2RewritePattern
                << "bVectorLen: " << bVectorLen << "\n"
                << "vectorTiebreaker: " << vectorTiebreaker << "\n");
 
-    // Vectorization lengths evenly devide *CopyPerThread, this is safe.
-    int64_t aCopyKPerThread = aVectorDim == GemmDimension::K
-                                  ? aVectorLen
-                                  : aCopyPerThread / aVectorLen;
-    int64_t copyMPerThread = aVectorDim == GemmDimension::MorN
-                                 ? aVectorLen
-                                 : aCopyPerThread / aVectorLen;
-    int64_t bCopyKPerThread = bVectorDim == GemmDimension::K
-                                  ? bVectorLen
-                                  : bCopyPerThread / bVectorLen;
-    int64_t copyNPerThread = bVectorDim == GemmDimension::MorN
-                                 ? bVectorLen
-                                 : bCopyPerThread / bVectorLen;
+    int64_t aCopyKPerThread = 0;
+    int64_t copyMPerThread = 0;
+    if (aVectorDim == GemmDimension::K) {
+      aCopyKPerThread = std::min(aVectorLen, kPerBlock);
```

Comment:
Huh. I guess it's not surprising you can hit this case - I guess we never had it with non-xdlops because of their tuning space.

Might be worth doing this to non-xdlops too?

Reply:
Sounds good, will generalize in a subroutine so both xdlops/nonxdlops can use it.

Reply:
Addressed in ef3f98f23bd6657d79daf0fbbc5b73800c5723f6